hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetadataContext.java

```diff
@@ -274,9 +274,10 @@ public Map<String, IdentifiableDomainType<?>> getIdentifiableTypesByName() {
 			attribute = factoryFunction.apply( entityType, genericProperty );
 			if ( !property.isGeneric() ) {
 				final PersistentAttribute<X, ?> concreteAttribute = factoryFunction.apply( entityType, property );
-				@SuppressWarnings("unchecked")
-				final AttributeContainer<X> attributeContainer = (AttributeContainer<X>) entityType;
-				attributeContainer.getInFlightAccess().addConcreteGenericAttribute( concreteAttribute );
+				if (nonNull(concreteAttribute)) {
+					@SuppressWarnings("unchecked") final AttributeContainer<X> attributeContainer = (AttributeContainer<X>) entityType;
+					attributeContainer.getInFlightAccess().addConcreteGenericAttribute(concreteAttribute);
```

Comment:
Please respect our code style.

```suggestion
					attributeContainer.getInFlightAccess().addConcreteGenericAttribute( concreteAttribute );
```